drivers/serial/uart_mcux_lpuart.c

```diff
@@ -1017,8 +1020,30 @@ static int mcux_lpuart_configure_init(const struct device *dev, const struct uar
 		/* Set the LPUART into loopback mode */
 		config->base->CTRL |= LPUART_CTRL_LOOPS_MASK;
 		config->base->CTRL &= ~LPUART_CTRL_RSRC_MASK;
+	} else if (config->single_wire) {
+		/* Enable the single wire / half-duplex mode, only possible when
+		 * loopback is disabled. We need a critical section to prevent
+		 * the UART firing an interrupt during mode switch
+		 */
+		unsigned int key = irq_lock();
+
+		config->base->CTRL |= (LPUART_CTRL_LOOPS_MASK | LPUART_CTRL_RSRC_MASK);
+		irq_unlock(key);
+	} else {
+#ifdef LPUART_CTRL_TXINV
+		/* Only invert TX in full-duplex mode */
+		if (config->tx_invert) {
+			config->base->CTRL |= LPUART_CTRL_TXINV(1);
+		}
+#endif
 	}
 
+#ifdef LPUART_STAT_RXINV
```

Comment:
Per your comment above (`/* Only invert TX in full-duplex mode */`), should this be moved to `else` case for single-wire mode only.

Reply:
I think the point is only one of the wires is used in single wire mode